tests/test_util.py

```diff
@@ -1510,6 +1511,29 @@ def test_introspection_signal_handler(capfd):
     assert log_msg in err
 
 
+@pytest.mark.parametrize('dockerignore_exists', [True, False])
+def test_allow_repo_dir_in_dockerignore(tmpdir, dockerignore_exists):
+    docker_ignore_file = os.path.join(str(tmpdir), DOCKERIGNORE)
+
+    ignore_content = """\
+# Ignore everything so we can just have a whitelist of things to copy
+** / *
+"""
+    added_line = "!%s\n" % RELATIVE_REPOS_PATH
+    if dockerignore_exists:
+        with open(docker_ignore_file, "w") as f:
+            f.write(ignore_content)
+
+    allow_repo_dir_in_dockerignore(tmpdir)
+
+    if dockerignore_exists:
+        with open(docker_ignore_file, "r") as f:
+            ignore_lines = f.readlines()
+
+        assert len(ignore_lines) == 3
+        assert ignore_lines[2] == added_line
```

Comment:
This seems too prescriptive. If all we need is for it to be the last line in the file, and for the existing content to still be in place, we should test those two things:
```
with open(docker_ignore_file, "r") as f:
    # Last line is the one we added.
    assert f.readlines()[-1] == added_line

with open(docker_ignore_file, "r") as f:
    # Previous content is still in place.
    assert f.read(len(ignore_content)) == ignore_content
```